MirServer/mir
# Support zwp_input_method_v2 and zwp_text_input_v3

This allows Squeekboard to auto-show, and show different layouts depending on the type of text field. This PR will be split into a number of pieces and merged separately.

To test:
- Run Mir like so: `WAYLAND_DISPLAY=mir ./bin/miral-shell --add-wayland-extension 'zwlr_layer_shell_v1:zwp_input_method_v2:zwp_virtual_keyboard_v1'`
- Make sure the GNOME OSK setting is enabled: `gsettings set org.gnome.desktop.a11y.applications screen-keyboard-enabled true`
- Run Squeekboard in the Mir session: `WAYLAND_DISPLAY=mir squeekboard`
- Select different applications to see the keyboard auto show/hide and show different content (for example it adds an extra row at the top in terminals)

You can test all the different types of input fields by running `tools/entry.py` in the squeekboard repo.

With the GNOME OSK set to false the keyboard can be manually shown with `busctl call --user sm.puri.OSK0 /sm/puri/OSK0 sm.puri.OSK0 SetVisible b true`, but it will not auto-show.
